crates/cli/src/modules/webhooks.rs

```diff
@@ -0,0 +1,60 @@
+use crate::help_strings::HelpStrings;
+use clap::{Args, Subcommand};
+use std::str;
+use tungstenite::{connect, Message};
+use url::Url;
+use serde_json::{from_str, to_string_pretty, Value};
+use colored::Colorize;
+
+
+/// Webhooks options and flags
+#[derive(Args)]
+#[clap(about = HelpStrings::Schema)]
+pub struct WebhooksOptions {
+    /// All the subcommands of the Webhooks cli
+    #[clap(subcommand)]
+    pub commands: WebhooksSubcommands,
+}
+
+/// Webhooks subcommands
+#[derive(Subcommand, Debug)]
+#[clap(about = HelpStrings::Webhooks)]
+pub enum WebhooksSubcommands {
+    /// Listen for webhooks on provided url
+    #[clap(about = HelpStrings::Webhooks)]
+    Listen {},
+}
+
+/// Listen to webhooks for an agentURL
+pub fn listen(agent_url: String) -> ! {
```

Comment:
I am not a fan of the `!` return in this case.

As I mentioned before, we should return a result so we can handle errors like we are doing in every submodule.

```suggestion
pub fn listen(agent_url: String) -> Result<()> {
```

This makes it very easy to handle errors in a consistent way across all the modules.

Using `unreachable!` and `unwrap` is extremely discouraged and there must be a very good reason for it to be in the codebase.


In this case we should just bubble-up the error and deal it in a general way.

Reply:
I do get your point - however, wondering whether Result is the right return type cos it just prints to void until ctrl-c so it's more like void or error?! Dunno (*newbie facepalm*)

Reply:
so not sure how Ok is returned

Reply:
Ok(()) should be the last line of your code and will be returned when the loop exits for any reason. `<C-c>` will short-circuit. 

Your `void or error` is exactly what a `Result<()>` means.

return nil, or void on Ok
return Error on Error